examples/decoding/plot_decoding_spoc_CMC.py

```diff
@@ -82,3 +82,8 @@
 plt.legend()
 mne.viz.tight_layout()
 plt.show()
+
+# Plot the contributions to the detected components (i.e., the forward model)
+layout = read_layout('CTF151.lay')
```

Comment:
Do you actually need to supply this? It suggests there is some deficiency in our `plot_patterns` / viz code that it does not auto-detect the layout.

We can fix it in a separate PR if so

Reply:
The principle I try to apply is to make each example focused on doing one thing (or small set of things) as clearly and simply as possible. If we start to show all options for functions in examples it dilutes the point. Thus showing how to use `layout` properly, in cases where you actually need to, can/should be done in some other example. Make sense?

Feel free to remove it from the EEG CSP example, too, if it still works without the arg there.

Reply:
Looks like EEG CSP needs it for the head position estimation; otherwise it fails with a runtime error "No digitization points found".  So, dodged a bullet there. ;)

Reply:
Removing `read_layout` from the SPoC example produces a runtime warning "Failed to locate mag channel positions from layout. Inferring channel positions from data".  The output looks the same either way, and from what I can tell from the Fieldtrip example, the channel names are identical either way.  Better to avoid the warning, though, I guess.  I can remove the channel type, since that's being detected.

Reply:
Yes we do not want warnings in examples. Sounds like a good plan